.github/workflows/spellcheck.yaml

```diff
@@ -0,0 +1,32 @@
+# Copyright (C) 2022 Daniel Stenberg, <daniel@haxx.se>, et al.
+#
+# SPDX-License-Identifier: curl
+#
+# Docs: https://github.com/UnicornGlobal/spellcheck-github-actions
+matrix:
```

Comment:
The workflows folder is reserved for actual workflow definitions. This YAML files is interpreted by GitHub and leads to invalid workflow runs, see: https://github.com/curl/curl/actions/workflows/spellcheck.yaml

They recommend to use `.github/scripts/` for workflow-related scripts, see https://docs.github.com/en/actions/learn-github-actions/essential-features-of-github-actions#adding-scripts-to-your-workflow